### ISSUE TITLE
Cannot decommission dead node

### ISSUE THREAD

**Issue by PiotrKozimor** [cockroachdb/cockroach#69137]
**Describe the problem**

Command to decommission node fails consistently:
```
$ kubectl exec -it cockroachdb-0 -- cockroach node decommission 3  --certs-dir=/cockroach/cockroach-certs
ERROR: server closed the connection.
Is this a CockroachDB node?

failed to connect to the node: initial connection heartbeat failed: rpc error: code = Unavailable desc = connection error: desc = "transport: authentication handshake failed: context deadline exceeded"
Failed running "node decommission"
command terminated with exit code 1
```
with following error on server side:
```
E210819 10:59:09.717440 51263 server/server.go:1914 ⋮ [n1,client=‹[::1]:34036›] serving SQL client conn: message size ‹352 MiB› bigger than maximum allowed message size ‹16 MiB›
```
I can list nodes with no problem:
```
$ kubectl exec -it cockroachdb-0 -- cockroach node ls  --certs-dir=/cockroach/cockroach-certs
  id
------
   1
   2
   3
   4
```

**To Reproduce**

1. Setup cluster with operator and local SSD on GKE.
2. Delete VM: `gcloud compute instances delete <instance>`
3. New node is added.
4. Old node is marked as dead.
5. Ranges are transferred to new node.
6. Try to decommission old node.

**Expected behavior**
Decommissioning succeeds. 

**Environment:**
 - CockroachDB version v20.2.14
 - Server OS: Container Optimized OS with containerd
 - Client app: `cockroach`

**Comment by RichardJCai** [comment]
I think generally KV deals with node decommissioning. Putting on the KV radar for now, let me know if this should've been put elsewhere.

**Comment by tbg** [comment]
Hi @PiotrKozimor, this seems odd. Are you reliably seeing that SQL message when invoking the decommissioning command? That message comes from handling SQL connections, which the decommission cli command does not initiate. Independently, it is odd that the decommissioning cli doesn't manage to establish a connection but a `node ls` following the exact same pattern does.

**Comment by knz** [comment]
@PiotrKozimor the SQL-related message and the decommission problem are completely unrelated.

The decommission problem indicates that the `cockroach decommission` command was unable to log into the crdb cluster. The particular error indicates that the system ranges supporting the `system.users` table or a related table is currently unavailable and/or overloaded.

Are you trying to decommission a node while the cluster is unhealthy? Generally, the decommission process can only succeed if the system ranges are healthy.

It seems to me, based on the few symptoms you have shared, that you need to provide more data about your situation before we can help you. It might also help to reach out to our technical support or or community Slack server so you can receive interactive help.

**Comment by knz** [comment]
@lunevalex isn't this addressed by your ongoing PR #69972? Doesn't it make it a KV issue?

**Comment by tbg** [comment]
I'm not sure what connects the two, Raphael. Anyway, let's close this? The symptoms are consistent with a networking issue or misconfiguration (such as pointing the CLI at the SQL port). Piotr, if you're still here try to get output from `cockroach debug gossip-values` run against the node (instead of `node decommission X`). This also uses a GRPC connection and should similarly fail, confirming that this has nothing to do with the decommissioning process itself.

**Comment by PiotrKozimor** [comment]
Looks like command works just right. I guess I did simple mistake of pointing to wrong port (26257 instead of 26258). If that's the case, I find [documentation](https://www.cockroachlabs.com/docs/stable/remove-nodes.html#step-1-mark-the-dead-node-as-decommissioned) misleading. Error message doesn't help too :/

```
$ kubectl get svc cockroachdb-public
NAME                 TYPE        CLUSTER-IP    EXTERNAL-IP   PORT(S)                        AGE
cockroachdb-public   ClusterIP   10.40.9.143   <none>        26258/TCP,8080/TCP,26257/TCP   29h
```

```
$ kubectl exec -it cockroachdb-client-secure -- sh
sh-4.4# cockroach debug --certs-dir /cockroach/cockroach-certs/ --host cockroachdb-public:26258 gossip-values
"cluster-id": 005ce68a-5dae-4346-90f5-b2f0aefebeea
"distsql-draining:1": {Draining:false}
"distsql-draining:2": {Draining:false}
"distsql-draining:3": {Draining:false}
"distsql-version:1": {Version:46 MinAcceptedVersion:44}
"distsql-version:2": {Version:46 MinAcceptedVersion:44}
"distsql-version:3": {Version:46 MinAcceptedVersion:44}
"first-range": r1:/{Min-System/NodeLiveness} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=4]
"gossip-clients:1": 3
"gossip-clients:2":
"gossip-clients:3": 2
"liveness:1": liveness(nid:1 epo:6 exp:1633451307.129140753,0)
"liveness:2": liveness(nid:2 epo:6 exp:1633451306.350416558,0)
"liveness:3": liveness(nid:3 epo:7 exp:1633451308.013740778,0)
"node:1": {NodeID:1 Address:cockroachdb-0.cockroachdb.default:26258 Attrs: Locality: ServerVersion:21.1 BuildTag:v21.1.9 StartedAt:1633343833391547993 LocalityAddress:[] ClusterName: SQLAddress:cockroachdb-0.cockroachdb.default:26257}
"node:2": {NodeID:2 Address:cockroachdb-2.cockroachdb.default:26258 Attrs: Locality: ServerVersion:21.1 BuildTag:v21.1.9 StartedAt:1633343747162706027 LocalityAddress:[] ClusterName: SQLAddress:cockroachdb-2.cockroachdb.default:26257}
"node:3": {NodeID:3 Address:cockroachdb-1.cockroachdb.default:26258 Attrs: Locality: ServerVersion:21.1 BuildTag:v21.1.9 StartedAt:1633343798281005446 LocalityAddress:[] ClusterName: SQLAddress:cockroachdb-1.cockroachdb.default:26257}
"sentinel": 005ce68a-5dae-4346-90f5-b2f0aefebeea
"store:1": {StoreID:1 Attrs: Node:{NodeID:1 Address:cockroachdb-0.cockroachdb.default:26258 Attrs: Locality: ServerVersion:21.1 BuildTag:v21.1.9 StartedAt:1633343833391547993 LocalityAddress:[] ClusterName: SQLAddress:cockroachdb-0.cockroachdb.default:26257} Capacity:disk (capacity=98 GiB, available=97 GiB, used=807 MiB, logicalBytes=2.9 GiB), ranges=122, leases=36, queries=10.41, writes=293.18, bytesPerReplica={p10=0.00 p25=0.00 p50=96.00 p75=8380.00 p90=388529.00 pMax=467136615.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.60 pMax=42.95}}
"store:2": {StoreID:2 Attrs: Node:{NodeID:2 Address:cockroachdb-2.cockroachdb.default:26258 Attrs: Locality: ServerVersion:21.1 BuildTag:v21.1.9 StartedAt:1633343747162706027 LocalityAddress:[] ClusterName: SQLAddress:cockroachdb-2.cockroachdb.default:26257} Capacity:disk (capacity=98 GiB, available=97 GiB, used=808 MiB, logicalBytes=2.9 GiB), ranges=122, leases=46, queries=4.01, writes=293.56, bytesPerReplica={p10=0.00 p25=0.00 p50=96.00 p75=8380.00 p90=388529.00 pMax=467096575.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.60 pMax=43.02}}
"store:3": {StoreID:3 Attrs: Node:{NodeID:3 Address:cockroachdb-1.cockroachdb.default:26258 Attrs: Locality: ServerVersion:21.1 BuildTag:v21.1.9 StartedAt:1633343798281005446 LocalityAddress:[] ClusterName: SQLAddress:cockroachdb-1.cockroachdb.default:26257} Capacity:disk (capacity=98 GiB, available=97 GiB, used=978 MiB, logicalBytes=2.9 GiB), ranges=122, leases=40, queries=7.80, writes=294.19, bytesPerReplica={p10=0.00 p25=0.00 p50=96.00 p75=8380.00 p90=388529.00 pMax=467086565.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.60 pMax=43.08}}
"system-db": omitted
"table-stat-added:100": 2021-09-25 21:54:17.426636696 +0000 UTC
...

sh-4.4# cockroach debug --certs-dir /cockroach/cockroach-certs/ --host cockroachdb-public gossip-values
ERROR: server closed the connection.
Is this a CockroachDB node?

initial connection heartbeat failed: rpc error: code = Unavailable desc = connection error: desc = "transport: authentication handshake failed: context deadline exceeded"
Failed running "debug gossip-values"
```

**Comment by ADustyOldMuffin** [comment]
I'll revive an old issue and say the wrong port was also an issue for me, error given back doesn't really help much and there's no documentation as to which port to use.
